test/functional/test_framework/mininode.py

```diff
@@ -366,7 +366,7 @@ def on_version(self, message):
 
     def wait_for_disconnect(self, timeout=60):
         test_function = lambda: not self.is_connected
-        wait_until(test_function, timeout=timeout, lock=mininode_lock)
+        wait_until(test_function, timeout=timeout, lock=mininode_lock, factor=self.factor)
```

Comment:
```suggestion
        self.wait_until(test_function, timeout=timeout)
```

Could this be a member function, so that the lock and factor members don't have to be passed each time?